es6/experiment.js

```diff
@@ -7,27 +7,17 @@ export default function provideExperiment(Assignment) {
       this._exposureLogged = false;
       this._salt = null;
       this._inExperiment = true;
-
-      this.name = this.getDefaultExperimentName();
```

Comment:
I think you make a good point here. The steps to creating an experiment / namespace in planout.js feel, in my opinion, a bit cumbersome.

Lately, I've been wondering whether planout.js needs to utilize classical inheritance like it's python & java siblings. The fact of the matter is that javascript is not python or java. And so the question arises, how closely do the implementation details need to match between each languages planout port if the core assignment algorithm remains consistent?

Something like this would be a pretty major change, so it's something I want to think about more and hear other opinions on before making any moves. As it stands, I think that there's a high level of verbosity that can be trimmed away when creating experiments & namespaces.

Do you have a strong opinion one way or the other?